usim/_basics/resource.py

```diff
@@ -0,0 +1,208 @@
+from typing import TypeVar, Generic, Optional, Type
+
+from .._core.loop import __LOOP_STATE__
+from ._resource_level import __specialise__, ResourceLevels
+from .tracked import Tracked
+
+
+T = TypeVar('T')
+
+
+class BaseResources(Generic[T]):
+    """
+    Internal base class for resource types
+    """
+    _levels_type = None  # type: Type[ResourceLevels[T]]
+    _available = None  # type: Tracked[ResourceLevels[T]]
+
+    @property
+    def levels(self) -> ResourceLevels[T]:
+        """Current levels of resources"""
+        return self._available.value
+
+    @property
+    def resources_type(self) -> Type[ResourceLevels[T]]:
```

Comment:
Why is it called resources instead of resource?

```suggestion
    def resource_type(self) -> Type[ResourceLevels[T]]:
```